### PR TITLE
Add real automated ci tests

### PR DESCRIPTION
**Pull request by echohack**
Signed-off-by: David Echo <echohack@users.noreply.github.com>

This PR adds automated tests for Linux for scaffolding-chef-infra. These tests will run on every PR and verify success or failure.

I've tried to replicate the CI environment and test this by hand, but there's a good chance that something in this PR might be off and cause the build to break. If that's the case I'll investigate and fix it right away. (It's hard to test the test system... inception all the way down).

Since this has a chance of breaking the build, I've kept the test case here minimal, and tested that out in my own Studio, as well as a chrooted studio that mimics that environment of the CI.

You can test this by hand, using

```
cd effortless
hab studio enter
build scaffolding-chef-infra
source results/last_build.env
build scaffolding-chef-infra/tests/user-linux-scaffolding-chef-infra
source results/last_build.env
./scaffolding-chef-infra/tests/test.sh ${pkg_ident}
```

Testing a package that consumes the scaffolding is a bit tricky. I've dubbed a new term here, a "user plan". Similar to a "user.toml" the user plan is a plan that consumes a scaffolding. I've named the test harness "user-<platform>-<package-name>" to allow us to variable-ize the automated test steps.

Overall I'm happy with this approach because it allows us to test the user plan in the same manner that it is getting used by our users. That allows us to create a battery of tests that hits the public API that we are creating.

`bats` and `pester` will allow us to test the build time API. Once we have some build time API tests in place, then we'll need to figure out how to test the runtime. We may need InSpec for this, we may not, but we'll cross that bridge when we come to it soon.

### CODE REVIEW

**Comment by echohack** [comment]
Note that I'm removing the `.bldr.toml` requirement from CI. This is super flakey as a lint check (we've had problems with it over in core-plans...) and it doesn't provide safety here since we only h ave two plans.

This will fail the ci, since you have to merge this CI check for the check to be removed, so an admin merge will be required because the test harnesses are plans themselves.

**Comment by echohack** [comment]
Closes #57

File: scaffolding-chef-infra/tests/user-linux/habitat/plan.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+pkg_name=user-linux
+pkg_origin=chef
+pkg_version="1.0.0"
+pkg_upstream_url="http://chef.io"
+pkg_scaffolding="ci/scaffolding-chef-infra"
+pkg_svc_user=("root")
+scaffold_policy_name="ci"
+scaffold_chef_client="chef/chef-client/14.13.11"
```
Comment (jerryaldrichiii) [inline scaffolding-chef-infra/tests/user-linux/habitat/plan.sh]: Any reason to pin here? I'd like to run latest so we can catch issues if/when they happen.

Comment (jerryaldrichiii) [reply scaffolding-chef-infra/tests/user-linux/habitat/plan.sh]: Same here https://github.com/chef/effortless/pull/100/files#diff-2ef69c17f3405c345463fc122527ec09R13

Comment (echohack) [reply scaffolding-chef-infra/tests/user-linux/habitat/plan.sh]: Pinning here is part of the test. We need to test that pinning works.

This user plan is a test harness, so it's ok to pin.

I'll add a comment that outlines this.

Comment (TheLunaticScripter) [reply scaffolding-chef-infra/tests/user-linux/habitat/plan.sh]: Okay so if I understand then we would also have a test that doesn't have the client pinned if we wanted to test the latest clients?

**Pull request merged**